File: backend/internal/model/task.go
package model

import "time"

// 任务类型
const (
	TypeDecompress = "decompress"
	TypeCompress   = "compress"
)

// 任务状态
const (
	StatusPending   = "pending"
	StatusRunning   = "running"
	StatusSucceeded = "succeeded"
	StatusFailed    = "failed"
	StatusCancelled = "cancelled"
)

// Task 对应一个文件的处理任务（解压或压缩）。
type Task struct {
	ID               uint       `gorm:"primaryKey" json:"id"`
	Type             string     `gorm:"size:16;index" json:"type"`
	SourcePath       string     `gorm:"size:1024;index" json:"source_path"`
	TargetPath       string     `gorm:"size:1024" json:"target_path"`
	TempPath         string     `gorm:"size:1024" json:"temp_path"`
	Status           string     `gorm:"size:16;index" json:"status"`
	ProgressPercent  int        `json:"progress_percent"`
	ProcessedBytes   int64      `json:"processed_bytes"`
	TotalBytes       int64      `json:"total_bytes"`
	Error            string     `gorm:"size:2048" json:"error"`
	// CompressionLevel 记录该压缩任务执行时实际使用的压缩效率档位
	// （fastest/fast/normal/slow）；解压任务不涉及压缩效率变更，此项为空。
	CompressionLevel string `gorm:"size:16" json:"compression_level"`
	// RequeueCount 记录该任务因「服务中断」被自动重新排队的次数（用于限制无限重排）。
	RequeueCount     int        `json:"requeue_count"`
	CreatedAt        time.Time  `gorm:"index" json:"created_at"`
	StartedAt        *time.Time `json:"started_at"`
	CompletedAt      *time.Time `gorm:"index" json:"completed_at"`
}

func (Task) TableName() string { return "tasks" }

File: frontend/src/utils/task.ts
/** 任务状态/类型展示映射（TasksPage 与 TaskDetailPage 共用）。 */

/** 运行中任务的轮询刷新间隔（毫秒）。 */
export const POLL_INTERVAL = 2000

/** 压缩效率档位 → 中文名（配置页下拉框与顶栏展示共用，保证文案一致）。 */
export const COMPRESSION_LABELS = {
  fastest: '特快',
  fast: '快',
  normal: '中',
  slow: '慢'
} as const

export type CompressionLevel = keyof typeof COMPRESSION_LABELS

/** 压缩效率中文名；未知取值回落"中"（与后端默认档位一致）。 */
export function compressionLabel(level: string | null | undefined): string {
  if (!level) return COMPRESSION_LABELS.normal
  return COMPRESSION_LABELS[level as CompressionLevel] ?? COMPRESSION_LABELS.normal
}

export function statusColor(status: string): string {
  switch (status) {
    case 'pending': return 'grey-6'
    case 'running': return 'blue-7'
    case 'succeeded': return 'green-7'
    case 'failed': return 'red-7'
    default: return 'grey-6'
  }
}

export function statusLabel(status: string): string {
  switch (status) {
    case 'pending': return '待处理'
    case 'running': return '执行中'
    case 'succeeded': return '成功'
    case 'failed': return '失败'
    default: return status
  }
}

export function typeLabel(type: string): string {
  return type === 'decompress' ? '解压' : '压缩'
}
